src/client/app/components/maps/MapViewComponent.tsx

```diff
@@ -2,436 +2,91 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import * as moment from 'moment';
 import * as React from 'react';
-import { FormattedMessage, injectIntl, WrappedComponentProps } from 'react-intl';
-import { Link } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { FormattedMessage} from 'react-intl';
 import { Button } from 'reactstrap';
+import * as moment from 'moment';
 import { CalibrationModeTypes, MapMetadata } from '../../types/redux/map';
-import { showErrorNotification } from '../../utils/notifications';
 import { hasToken } from '../../utils/token';
+import '../../styles/card-page.css';
+import EditMapModalComponent from './EditMapModalComponent';
+
 
 interface MapViewProps {
-	// The ID of the map to be displayed
 	id: number;
```

Comment:
This is a big picture comment that I am putting here. In a conversion to the Redux Toolkit, most props and all the containers should go away. (For maps some will likely remain for the new map/calibration until that is converted/updated. I have not tried but I think it will be okay to remove the others while leaving these.) These still exist and need to be removed. I suspect the issue with needing a save on the card page rather than each edit modal relates to this. See other pages for the general idea and this should mean you can pass just the needed map state (actually id is good enough but that is not what most pages do). This is going to lead to non-trivial changes in the code so I'm not going to do a full review until this is done. If you need help with this then work with your team and/or contact me.

Reply:
This has been addressed with Brandon's most recent commits